crates/frontend/src/api.rs

```
@@ -528,3 +530,41 @@ pub async fn resolve_config(
         Err(e) => Err(e.to_string()),
     }
 }
+
+pub async fn execute_autocomplete_function(
+    name: &str,
+    value: &str,
+    environment: &Value,
+    fn_name: &String,
+    tenant: &String,
+    org_id: &String,
+) -> Result<Vec<String>, ServerFnError> {
+    let host = use_host_server();
+    let client = reqwest::Client::new();
+    let url = format!("{}/function/{}/PUBLISHED/test", host, fn_name);
+    let err_handler = |e: reqwest::Error| ServerFnError::new(e.to_string());
+    let payload = FunctionExecutionRequest::AutocompleteFunctionRequest {
+        name: name.to_owned(),
+        prefix: value.to_owned(),
+        environment: environment.clone(),
+    };
+    let response = client
+        .put(url)
+        .header("x-org-id", org_id)
+        .header("x-tenant", tenant)
+        .json(&payload)
+        .send()
+        .await
+        .map_err(err_handler)?
+        .json::<FunctionExecutionResponse>()
+        .await
+        .map_err(err_handler)?;
+    let result = response
+        .fn_output
+        .as_array()
+        .unwrap()
+        .iter()
+        .map(|v| v.as_str().unwrap().to_string())
```

`.filter_map(|v| v.as_str().map(String::from))`

---

I want to fail if an element is not a string, rather than ignoring it

---

that is not how you can handle this, you would need to send back an error from here and show the error somewhere probably as a snackbar

using unwrap simply crashes the frontend application giving feedback, so this technically worse than ignoring the error